components/ActionSheet/tests/unit/MDCActionSheetTest.m

```diff
@@ -204,4 +218,142 @@ - (void)testSetNilTitleAndMessageColor {
                         [UIColor.blackColor colorWithAlphaComponent:0.6f]);
 }
 
+#pragma mark - Opening height
+
+- (void)addActions:(NSUInteger)actions {
+  for (NSUInteger i = 0; i < actions; ++i) {
+    MDCActionSheetAction *action = [MDCActionSheetAction actionWithTitle:@"Title"
+                                                                   image:nil
+                                                                 handler:nil];
+    [self.actionSheet addAction:action];
+  }
+}
+
+- (void)testOpenningHeightWithNoHeader {
+  // Given
+  CGFloat fakeHeight = 500;
+  self.actionSheet.view.bounds = CGRectMake(0, 0, 200, fakeHeight);
+
+  // When
+  [self addActions:100];
+  [self.actionSheet.view setNeedsLayout];
+  [self.actionSheet.view layoutIfNeeded];
+
+  // Then
+  CGFloat headerHeight = CGRectGetHeight(self.actionSheet.header.frame);
+  CGFloat expectedHeight = [self.actionSheet openingSheetHeight];
+  CGFloat expectedMinusHeader = expectedHeight - headerHeight;
+  CGFloat cellHeight =
+      self.actionSheet.tableView.contentSize.height / (CGFloat)self.actionSheet.actions.count;
+  cellHeight = MDCCeil(cellHeight);
+  CGFloat halfCellHeight = cellHeight * 0.5f;
+  // Action sheet should show half of the allowed actions but the full last action
+  XCTAssertEqual(fmod(expectedMinusHeader, halfCellHeight), 0);
+  XCTAssertNotEqual(fmod(expectedMinusHeader, cellHeight), 0);
```

Comment:
Tip: If we fetch the last and second-last cells and check their frame as projected onto the action sheet's bounds I think this logic might be a bit easier to read. `CGRectContainsRect` may be helpful here.

Reply:
If I am understanding you write then in the case where we have 100 cells we would get the 99 and 100th cell. Those cells are off the screen so that isn't what we want to test against.